tests/test_data_loader.py

```diff
@@ -39,4 +39,4 @@ def test_preprocess_data(mocker, mock_df: pd.DataFrame):
         index=[0, 1, 2],
     )
     expected_df.index.name = "id"
-    assert_frame_equal(data_loader.party_data, expected_df, check_less_precise=3)
+    assert_frame_equal(data_loader.party_data, expected_df, rtol=3)
```

Comment:
The test failed for me, but is that expected?

Reply:
yep, that's expected. this test will only pass if the pre-processing exercise has been completed successfully. the test passes for me with a local solution